client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/hooks/use-recipe.ts

```diff
@@ -167,8 +180,10 @@ const useRecipe = (
 			return;
 		}
 
-		const requestedDesign = allDesigns.designs.find(
-			( design ) => design.recipe?.slug === preselectedTheme || design.slug === preselectedTheme
+		const requestedDesign = allDesigns.designs.find( ( design ) =>
+			design.is_virtual
```

Comment:
I've seen this pattern multiple times.

Looking at the output of the `/starter-designs` endpoint for this theme:

```
{
    "slug": "creatio",
    "title": "Link in Bio: Anime",
    "description": "Share curated links to all your sites and social media profiles.",
    "recipe": {
        "slug": "creatio-link-in-bio-with-anime-background-2",
        "pattern_ids": [
            "8367-174455321"
        ],
        "header_pattern_ids": [],
        "footer_pattern_ids": [],
        "stylesheet": "pub\/creatio"
    },
    "is_virtual": true,
    ...
},
```

I think we got it the way around. Shouldn't it be the case that

- `slug` = `creatio-link-in-bio-with-anime-background-2`
- `recipe.slug` = `creatio`?

`design.slug` should be unique as this is used by Tracks events etc.

Maybe we should fix this in the future... (not for this PR) 😭

Reply:
Wait maybe `recipe` needs to be `creatio-link-in-bio-with-anime-background-2` because we want to give the recipe a name, but still the `design.slug` should be `creatio-link-in-bio-with-anime-background-2`, I'll take a look probably later 😄 

If not this will bite us again for sure sooner or later in the future 😅

Reply:
Yep, I agree the `design.slug` and `design.recipe.slug` should be the same 👍 

Let's confirm with @Automattic/t-rex to see whether it's reasonable 🙂

Reply:
> `design.slug` should be unique as this is used by Tracks events etc.

Correct, this is why we track a different slug for virtual themes:

https://github.com/Automattic/wp-calypso/blob/60d8f409319cfdbacf001bb3c7457fda78e1443e/client/landing/stepper/declarative-flow/internals/analytics/record-design.ts#L105

> `design.slug` and `design.recipe.slug` should be the same

This sounds like a good idea which would simplify some pieces of logic. I took a quick look and seems that there won't be unexpected side effects if `design.slug` suddenly changes to the recipe slug, we definitely need to take a deeper look to confirm this (there might some places where `design.slug` is expected to be the theme slug.